### ISSUE TITLE
Failed to load shaderpack when coloured lights is enabled

### ISSUE THREAD

**Issue by OKHI233** [sixthsurge/photon#132]
### What happened?

When coloured lights is enabled, the shaderpack fails to load. I'm using the commit 08bf5a0 version.
The shaderpack works fine if coloured lights is disabled otherwise.

### Minecraft Version

1.20.1

### Which shader mod are you using?

Iris

### Shader mod version

1.6.17

### Shader settings

```conf
#Tue Apr 09 00:34:15 HKT 2024
AIR_FOG_COLORED_LIGHT_SHAFTS=true
BLOCK_ENTITY_SHADOWS=true
COLORED_LIGHTS=true
INFO=3
NORMAL_MAPPING=true
SPECULAR_MAPPING=true
WATER_CAUSTICS=true
shadowMapResolution=4096
```

### Relevant log output

```text
[00:34:16] [Render thread/INFO]: Starting custom uniform resolving
[00:34:17] [Render thread/INFO]: Mixing client.MixinGuiMessageTag from #nochatreports:mixins/common/nochatreports.mixins.json into net.minecraft.class_7591
[00:34:17] [Render thread/INFO]: [CHAT] The shaderpack failed to load! Please report the error to the shader developer. Copy Info
[00:34:17] [Render thread/ERROR]: Failed to create shader rendering pipeline, disabling shaders!
net.coderbot.iris.gl.shader.ShaderCompileException: shadow_terrain_cutout.vsh: shadow_terrain_cutout.vsh: ERROR: 0:270: '||' :  wrong operand types: no operation '||' exists that takes a left-hand operand of type ' global 3-component vector of bool' and a right operand of type ' global 3-component vector of bool' (or there is no acceptable conversion)
ERROR: 0:270: 'all' : no matching overloaded function found 
ERROR: 0:270: '' : compilation terminated 
ERROR: 3 compilation errors.  No code generated.


	at net.minecraft.class_281.handler$cam000$iris$causeException(class_281.java:536) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_281.method_34416(class_281.java:68) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_281.method_1283(class_281.java:51) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_5944.method_34579(class_5944.java:242) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_5944.<init>(class_5944.java:185) ~[minecraft-1.20.1.i0:0/:?]
	at net.coderbot.iris.pipeline.newshader.ExtendedShader.<init>(ExtendedShader.java:84) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.pipeline.newshader.NewShaderTests.create(NewShaderTests.java:110) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.pipeline.newshader.NewWorldRenderingPipeline.createShadowShader(NewWorldRenderingPipeline.java:760) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.pipeline.newshader.NewWorldRenderingPipeline.createShadowShader(NewWorldRenderingPipeline.java:735) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.pipeline.newshader.NewWorldRenderingPipeline.lambda$new$12(NewWorldRenderingPipeline.java:425) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.pipeline.newshader.ShaderMap.<init>(ShaderMap.java:21) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.pipeline.newshader.NewWorldRenderingPipeline.<init>(NewWorldRenderingPipeline.java:421) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.Iris.createPipeline(Iris.java:653) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.pipeline.PipelineManager.preparePipeline(PipelineManager.java:35) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.Iris.reload(Iris.java:601) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.apiimpl.IrisApiV0ConfigImpl.setShadersEnabledAndApply(IrisApiV0ConfigImpl.java:28) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.gui.screen.ShaderPackScreen.applyChanges(ShaderPackScreen.java:586) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.gui.screen.ShaderPackScreen.method_25419(ShaderPackScreen.java:538) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.gui.screen.ShaderPackScreen.lambda$init$2(ShaderPackScreen.java:241) ~[iris-1.6.17.i0:0/:?]
	at net.minecraft.class_4185.method_25306(class_4185.java:94) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_4264.method_25348(class_4264.java:56) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_339.method_25402(class_339.java:189) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_4069.method_25402(class_4069.java:38) ~[minecraft-1.20.1.i0:0/:?]
	at net.coderbot.iris.gui.screen.ShaderPackScreen.method_25402(ShaderPackScreen.java:198) ~[iris-1.6.17.i0:0/:?]
	at net.minecraft.class_312.method_1611(class_312.java:98) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_437.method_25412(class_437.java:409) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_312.method_1601(class_312.java:98) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_312.method_22686(class_312.java:169) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_1255.execute(class_1255.java:102) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_312.method_22684(class_312.java:169) ~[minecraft-1.20.1.i0:0/:?]
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:43) ~[lwjgl-glfw-3.3.1.jar:build 7]
	at org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.3.1.jar:build 7]
	at org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3403) ~[lwjgl-glfw-3.3.1.jar:build 7]
	at com.mojang.blaze3d.systems.RenderSystem.pollEvents(RenderSystem.java:201) ~[minecraft-1.20.1.i0:0/:?]
	at com.mojang.blaze3d.systems.RenderSystem.flipFrame(RenderSystem.java:219) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_1041.method_15998(class_1041.java:288) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_310.method_1523(class_310.java:1241) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_310.method_1514(class_310.java:802) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.client.main.Main.main(Main.java:250) ~[quilt-loader-0.25.0-beta.3-1.20.1.jar:?]
	at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
	at org.quiltmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:551) ~[quilt-loader-0.25.0-beta.3.jar:?]
	at org.quiltmc.loader.impl.launch.knot.Knot.launch(Knot.java:84) ~[quilt-loader-0.25.0-beta.3.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:28) ~[quilt-loader-0.25.0-beta.3.jar:?]
```

**Comment by sixthsurge** [comment]
will try to fix

**Comment by OKHI233** [comment]
Unfortunately this is still an issue. :(
I don't know this is actually a hardware specific issue until I see commit 7860b9e. I am using AMD RX 7600, driver version 24.3.1, if it helps.
```
[05:56:54] [Render thread/INFO]: Starting custom uniform resolving
[05:56:55] [Render thread/INFO]: Mixing client.MixinGuiMessageTag from #nochatreports:mixins/common/nochatreports.mixins.json into net.minecraft.class_7591
[05:56:55] [Render thread/INFO]: [CHAT] The shaderpack failed to load! Please report the error to the shader developer. Copy Info
[05:56:55] [Render thread/ERROR]: Failed to create shader rendering pipeline, disabling shaders!
net.coderbot.iris.gl.shader.ShaderCompileException: shadow_terrain_cutout.vsh: shadow_terrain_cutout.vsh: ERROR: 0:270: 'return' : type does not match, or is not convertible to, the function's return type 
ERROR: 0:270: '' : compilation terminated 
ERROR: 2 compilation errors.  No code generated.


	at net.minecraft.class_281.handler$cam000$iris$causeException(class_281.java:536) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_281.method_34416(class_281.java:68) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_281.method_1283(class_281.java:51) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_5944.method_34579(class_5944.java:242) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_5944.<init>(class_5944.java:185) ~[minecraft-1.20.1.i0:0/:?]
	at net.coderbot.iris.pipeline.newshader.ExtendedShader.<init>(ExtendedShader.java:84) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.pipeline.newshader.NewShaderTests.create(NewShaderTests.java:110) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.pipeline.newshader.NewWorldRenderingPipeline.createShadowShader(NewWorldRenderingPipeline.java:760) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.pipeline.newshader.NewWorldRenderingPipeline.createShadowShader(NewWorldRenderingPipeline.java:735) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.pipeline.newshader.NewWorldRenderingPipeline.lambda$new$12(NewWorldRenderingPipeline.java:425) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.pipeline.newshader.ShaderMap.<init>(ShaderMap.java:21) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.pipeline.newshader.NewWorldRenderingPipeline.<init>(NewWorldRenderingPipeline.java:421) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.Iris.createPipeline(Iris.java:653) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.pipeline.PipelineManager.preparePipeline(PipelineManager.java:35) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.Iris.reload(Iris.java:601) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.apiimpl.IrisApiV0ConfigImpl.setShadersEnabledAndApply(IrisApiV0ConfigImpl.java:28) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.gui.screen.ShaderPackScreen.applyChanges(ShaderPackScreen.java:586) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.gui.screen.ShaderPackScreen.method_25419(ShaderPackScreen.java:538) ~[iris-1.6.17.i0:0/:?]
	at net.coderbot.iris.gui.screen.ShaderPackScreen.lambda$init$2(ShaderPackScreen.java:241) ~[iris-1.6.17.i0:0/:?]
	at net.minecraft.class_4185.method_25306(class_4185.java:94) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_4264.method_25348(class_4264.java:56) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_339.method_25402(class_339.java:189) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_4069.method_25402(class_4069.java:38) ~[minecraft-1.20.1.i0:0/:?]
	at net.coderbot.iris.gui.screen.ShaderPackScreen.method_25402(ShaderPackScreen.java:198) ~[iris-1.6.17.i0:0/:?]
	at net.minecraft.class_312.method_1611(class_312.java:98) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_437.method_25412(class_437.java:409) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_312.method_1601(class_312.java:98) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_312.method_22686(class_312.java:169) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_1255.execute(class_1255.java:102) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_312.method_22684(class_312.java:169) ~[minecraft-1.20.1.i0:0/:?]
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:43) ~[lwjgl-glfw-3.3.1.jar:build 7]
	at org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.3.1.jar:build 7]
	at org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3403) ~[lwjgl-glfw-3.3.1.jar:build 7]
	at com.mojang.blaze3d.systems.RenderSystem.pollEvents(RenderSystem.java:201) ~[minecraft-1.20.1.i0:0/:?]
	at com.mojang.blaze3d.systems.RenderSystem.flipFrame(RenderSystem.java:219) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_1041.method_15998(class_1041.java:288) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_310.method_1523(class_310.java:1241) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.class_310.method_1514(class_310.java:802) ~[minecraft-1.20.1.i0:0/:?]
	at net.minecraft.client.main.Main.main(Main.java:250) ~[quilt-loader-0.25.0-beta.3-1.20.1.jar:?]
	at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
	at org.quiltmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:551) ~[quilt-loader-0.25.0-beta.3.jar:?]
	at org.quiltmc.loader.impl.launch.knot.Knot.launch(Knot.java:84) ~[quilt-loader-0.25.0-beta.3.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:28) ~[quilt-loader-0.25.0-beta.3.jar:?]
```

**Comment by OKHI233** [comment]
Fixed by commit 96a9246, thank you for your fabulous shaderpack!
